contracts/FeralfileArtworkV4.sol

```diff
@@ -10,60 +10,216 @@ import "./Authorizable.sol";
 import "./UpdateableOperatorFilterer.sol";
 
 contract FeralfileExhibitionV4 is
-    IERC165,
-    ERC721Enumerable,
+    ERC721,
     Authorizable,
     UpdateableOperatorFilterer
 {
     using Strings for uint256;
 
+    struct Artwork {
+        uint256 seriesId;
+        uint256 tokenId;
+    }
+
+    struct MintData {
+        uint256 seriesId;
+        uint256 tokenId;
+        address minter;
+    }
+
     // version code of contract
     string public constant codeVersion = "FeralfileExhibitionV4";
 
     // burnable
-    bool public isBurnable;
+    bool private _burnable;
 
     // bridgeable
-    bool public isBridgeable;
+    bool private _bridgeable;
+
+    // mintable
+    bool private _mintable = true;
 
     // token base URI
     string internal _tokenBaseURI;
 
     // contract URI
     string private _contractURI;
 
+    // signer
+    address private _signer;
+
+    // series max supplies
+    mapping(uint256 => uint256) internal _seriesMaxSupplies;
+
+    // series total supplies
+    mapping(uint256 => uint256) internal _seriesTotalSupplies;
+
+    // total supply
+    uint256 private _totalSupply;
+
+    // all artworks
+    mapping(uint256 => Artwork) internal _allArtworks;
+
     constructor(
         string memory name_,
         string memory symbol_,
-        bool isBurnable_,
-        bool isBridgeable_
+        address signer_,
+        bool burnable_,
+        bool bridgeable_,
+        uint256[] memory seriesIds_,
+        uint256[] memory seriesMaxSupplies_
     ) ERC721(name_, symbol_) {
-        isBurnable = isBurnable_;
-        isBridgeable = isBridgeable_;
+        // validations
+        require(
+            bytes(name_).length > 0,
+            "FeralfileExhibitionV4: name_ is empty"
+        );
+        require(
+            bytes(symbol_).length > 0,
+            "FeralfileExhibitionV4: symbol_ is empty"
+        );
+        require(
+            signer_ != address(0),
+            "FeralfileExhibitionV4: signer_ is zero address"
+        );
+        require(
+            seriesIds_.length > 0,
+            "FeralfileExhibitionV4: seriesIds_ is empty"
+        );
+        require(
+            seriesMaxSupplies_.length > 0,
+            "FeralfileExhibitionV4: _seriesMaxSupplies is empty"
+        );
```

Comment:
1/1 is not a series, should we move that rule to contract level?
just bring up the discussion, I'm ok to put a series for 1/1 on contract/FF DB level but hiding from UI :)

Reply:
@lpopo0856 How can you deal with 1/1 without series?